Cron/DeleteBackups.php

```diff
@@ -0,0 +1,206 @@
+                            ['method' => CronConfig::OLDER_THAN, 'files' => $filesOlderThan]
+                        );
+                    }
+                }
+            }
+
+        }
+    }
+
+    /**
+     * @param $directoryIterator
+     * @param $now
+     * @return array
+     */
+    private function findAndSortFilesByPeriods($directoryIterator, $now): array
+    {
+        $files = $filesByWeeks = $filesByMonths = $filesByYears = [];
+
+        foreach (new \RecursiveIteratorIterator($directoryIterator) as $file) {
+            $filePath = $file->getPathname();
+            if (strpos($filePath, CronConfig::HISTORY_DIR)) {
+
+                if ($now - filemtime($filePath) >= CronConfig::WEEK && $now - filemtime($filePath) < CronConfig::MONTH) {
+                    $filesByWeeks[] = $filePath;
+                }
+
+                if ($now - filemtime($filePath) >= CronConfig::MONTH && $now - filemtime($filePath) < CronConfig::YEAR) {
+                    $filesByMonths[] = $filePath;
+                }
+
+                if ($now - filemtime($filePath) >= CronConfig::YEAR) {
+                    $filesByYears[] = $filePath;
+                }
+            }
+        }
+
+        if (count($filesByWeeks)) {
+            $files['week']['files'] = $filesByWeeks;
+            $files['week']['period'] = CronConfig::WEEK;
+        }
+
+        if (count($filesByMonths)) {
+            $files['month']['files'] = $filesByMonths;
+            $files['month']['period'] = CronConfig::MONTH;
+        }
+
+        if (count($filesByYears)) {
+            $files['year']['files'] = $filesByYears;
+            $files['year']['period'] = CronConfig::YEAR;
+        }
+
+        return $files;
+    }
+
+    /**
+     * @param $files
+     * @param $period
+     * @return array
+     */
+    private function deleteFilesByPeriod($files, $period): array
+    {
+        usort($files, function ($a, $b) { return filemtime($b) - filemtime($a); });
+
+        $previousFile = null;
+        $deletedFiles = [];
+
+        foreach ($files as $file) {
+            if (!$previousFile) $previousFile = $file;
+
+            if ($previousFile !== $file) {
+                if (filemtime($previousFile) - $period >= filemtime($file)) {
+                    $previousFile = $file;
+                } else {
+                    $deletedFiles[] = $file;
+                    unlink($file);
+                }
+            }
+        }
+
+        return $deletedFiles;
+    }
+
+    /**
+     * @param $directoryIterator
+     * @param $now
+     * @return array
+     */
+    private function findFilesOlderThan($directoryIterator, $now): array
+    {
+        $period = $this->cronConfig->getPeriodType() * $this->cronConfig->getPeriodNumber();
+        $files = [];
+
+        foreach (new \RecursiveIteratorIterator($directoryIterator) as $file) {
+            $filePath = $file->getPathname();
+            if (strpos($filePath, CronConfig::HISTORY_DIR)) {
```

Comment:
Better to make two separate ifs:
` 
 if (strpos($filePath, CronConfig::HISTORY_DIR)) { continue }
if (rule)
 `